app/services/fgdc_update_service.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+class FgdcUpdateService
+  def initialize(file_set:)
+    @file_set = file_set
+  end
+
+  def insert_onlink(url:)
+    onlink = find_or_create_node("//idinfo/citation/citeinfo/onlink")
+    onlink.content = url
+    save_changes
+  end
+
+  private
+
+    def doc
+      @doc ||= Nokogiri::XML(file_object.read)
+    end
```

Comment:
Should errors in XML parsing be caught here?

Reply:
If an FGDC document makes to this point, it should not have parsing errors, but it's probably good practice to rescue them.

Reply:
@eliotjordan indeed addressed this thoroughly, my concern was not warranted (due to the nature of the ingestion process for XML files in this context)

Reply:
After talking through this with James and trying to mock this up in specs, it really seems that this isn't necessary. Malformed FGDC XML won't make it to this point, because Figgy won't properly characterize / ingest the file. Does that sound right @jrgriffiniii?